ProcessLib/Output.h

```diff
@@ -84,20 +84,20 @@ class Output
         MeshLib::IO::PVDFile pvd_file;
     };
 
-    Output(std::string output_directory, std::string prefix, bool const compress_output,
-           bool const output_nonlinear_iteration_results)
-        : _output_directory(std::move(output_directory)),
-          _output_file_prefix(std::move(prefix)),
-          _output_file_compression(compress_output),
-          _output_nonlinear_iteration_results(
-              output_nonlinear_iteration_results)
-    {}
+    Output(std::string output_directory, std::string prefix,
+           bool const compress_output, std::string const data_mode,
```

Comment:
Why not `std::string const& data_mode` instead of `std::string const data_mode`?

Reply:
This is the "take easy-to-move types by value, and move them" idiom @endJunction introduced some weeks ago. Personally I'd prefer the old-fashioned `std::string const&`...
Note: `std::string` here is not const!

Reply:
Forgot the `&`. For move idiom it must be non-const.

Reply:
OK. I confused `data_mode` with the other string arguments in my comment above. IMHO passing by value or by reference deliberately depending on some arbitrary notion of consistency is not the best option: IMHO either move idiom for all strings or for none.

Reply:
Move for all strings which are copied. I think that all cases were fixed in https://github.com/ufz/ogs/pull/1779 https://github.com/ufz/ogs/pull/1779/commits/88426db960647ea4fc24c2efde6981bb9121a09f.

Still (as changed in the comment above) It would be not optimal here, because there is no copy at all.